### ISSUE TITLE
latest docker: ReadContractCalls: cannot marshal response

### ISSUE THREAD

**Issue by xplat** [blockwatch-cc/tzindex#25]
```bash
curl http://<internal server with latest docker image>/explorer/contract/KT1TTtduU7q21hUpMPQQKERy1NYaCTr72Q4s/calls?since=BKokVkUFwZ3KTcPWFuRNJj9SBB7qnUMqNwNN1LWosWsBWW2zVZ2\&limit=1
```

```json
{
  "errors": [
    {
      "code": 1003,
      "status": 500,
      "message": "internal server error",
      "scope": "ReadContractCalls",
      "detail": "cannot marshal response",
      "reason": "json: error calling MarshalJSON for type *micheline.BigMapValue: micheline: type mismatch val_type=prim[Unit] type_code=list / type={\"annots\":[\"%disburse\"],\"args\":[{\"prim\":\"address\"}],\"prim\":\"list\"} -- value={\"prim\":\"Unit\"} / map=map[string]interface {}{}"
    }
  ]
}
```

To be more precise, 'latest' is image ID `5d6cb56838e6` ...

**Comment by xplat** [comment]
This also happens with the 'v6.0.2' tag on docker hub ... and the logs look like this:

```
2020/04/30 20:30:38 ERRO MICH RENDER ERROR
typ={"annots":["%disburse"],"args":[{"prim":"address"}],"prim":"list"}
val={"prim":"Unit"}
err=micheline: type mismatch val_type=prim[Unit] type_code=list / type={"annots":["%disburse"],"args":[{"prim":"address"}],"prim":"list"} -- value={"prim":"Unit"} / map=map[string]interface {}{}
2020/04/30 20:30:38 ERRO SRVR BW-8f496776e498edb4c6d78d0a9ecae47cc6df42ab Error sending response: json: error calling MarshalJSON for type *micheline.BigMapValue: micheline: type mismatch val_type=prim[Unit] type_code=list / type={"annots":["%disburse"],"args":[{"prim":"address"}],"prim":"list"} -- value={"prim":"Unit"} / map=map[string]interface {}{} in struct []*server.ExplorerOp{(*server.ExplorerOp)(0xc00e45f2c0)}
```

**Comment by echa** [comment]
Thanks. This bug seems fixed in our PRO and Cloud releases, but the relevant code there was heavily refactored. I'll see if we can backport this as hotfix, otherwise this will be fixed with the next OSS release 6.1

**Comment by echa** [comment]
I had a closer look. The underlying reason for this bug is that the entrypoint is not detected correctly. It's a general limitation of the way entrypoints are handled in the current release. This will be fixed with the next release v6.1.

As hotfix I added a check so the call does not fail entirely, but it still returns the wrong entrypoint and an empty/no parameter value. As workaround to still access the original parameters you can add the `prim=1` request argument and access the raw Michelson JSON.

**Comment by echa** [comment]
Closing this old issue. The problem should have been solved by v7/v8. We no longer fail returning any operations where storage data unwrapping fails. Instead the unwrapped storage contains an embedded error object plus the original type + value primitive trees.
